portal-impl/src/com/liferay/portlet/documentlibrary/service/impl/DLFileShortcutServiceImpl.java

```diff
@@ -68,6 +68,30 @@ public void deleteFileShortcut(long fileShortcutId) throws PortalException {
 		dlFileShortcutLocalService.deleteFileShortcut(fileShortcutId);
 	}
 
+	@Override
+	public void deleteFileShortcut(String externalReferenceCode, long groupId)
+		throws PortalException {
+
+		DLFileShortcut fileShortcut =
+			dlFileShortcutLocalService.
+				fetchDLFileShortcutByExternalReferenceCode(
+					externalReferenceCode, groupId);
+
+		if (fileShortcut != null) {
```

Comment:
good point @marcogalluzzi. But I was also going to ask if we don't want to get the shortcut with

```
dlFileShortcutPersistence.findByERC_G(
				externalReferenceCode, groupId)
```
so that it throws directly an exception

Reply:
hmm, this is a bit unrelated, but shouldn't a ServiceBuilder delete method have a return value, in this case DLFileShortcut?

Also, I've seen multiple types of implementations.
I can't see a definite preference on calling LocalService vs Persistence in this case, but shouldn't we call the LocalService so that a customised LocalServiceWrapper might wrap?

About directly throwing an Exception, should we expose the non-existence of a DLFileShortcut to a remote call or should we throw a Permission Excection?

Reply:
@balazssk, good question. Seems we have a mix of patterns..... (as usual). But the most used from what I see is getting the entity (which throws the exception) and then checking the permissions. But it also depends on the team that implemented it 🤣🤣🤣🤣🤣🤣🤣

Reply:
I see,
I've added https://github.com/liferay-content-management/liferay-portal/pull/5716/commits/6ff78170999523973cb5dfe8ecadfb3c2950c4cc to have the NoSuchFileException exposed. I think we should still use the LocalService instead of the Persistence in this case